src/subcommands/deploy.ts

```diff
@@ -130,6 +130,13 @@ async function deploy(opts: DeployOpts): Promise<void> {
   }
   projectSpinner.succeed(`Project: ${project.name}`);
 
+  if (!project!.hasProductionDeployment) {
+    wait("").start().info(
+      "Empty project detected, automatically pushing initial deployment to production (use --prod for further updates).",
```

Comment:
We can't make this assumption as there can be existing deployments in a project without productionDeployment being set. We need to rewrite this so that we first fetch deployments for a project and then check if deployments.is_empty()

Reply:
Thanks for the feedback, updated the logic that way.